core dump when using wayland output and no input devices

### mpv version and platform

Using Arch Linux ARM on a Raspberry Pi 4.

```
$ mpv --version
mpv 0.29.1 Copyright © 2000-2018 mpv/MPlayer/mplayer2 projects
 built on Sun Oct  6 21:40:18 UTC 2019
ffmpeg library versions:
   libavutil       56.31.100
   libavcodec      58.54.100
   libavformat     58.29.100
   libswscale      5.5.100
   libavfilter     7.57.100
   libswresample   3.5.100
ffmpeg version: n4.2.1
```

### Reproduction steps

do not have a mouse and keyboard attached and run `mpv` within `weston`.

I usually do this over ssh and ensuring that the `WAYLAND_DISPLAY` value is set.

`~/.config/weston.ini`:
```
[core]
require-input=false
idle-time=0

[shell]
locking=false
clock-format=none
background-color=0xFF000000
animation=none
close-animation=none
startup-animation=none
focus-animation=none
```

`~/.config/mpv/mpv.conf`:
```
fullscreen=yes
loop-playlist=inf
input-cursor=no
input-terminal=no
osc=no
osd-bar=no
terminal=yes
input-default-bindings=no
```

### Expected behavior

mpv runs in fullscreen in weston.

### Actual behavior

```
$ ./start
[playlist] Reading plaintext playlist.
Playing: /home/alarm/videos/1.x265.mp4
 (+) Video --vid=1 (*) (hevc 1280x720 24.000fps)
 (+) Audio --aid=1 --alang=eng (*) (aac 2ch 48000Hz)
error marshalling arguments for get_data_device (signature no): null value passed for arg 1
Error marshalling request: Invalid argument
./start: line 16:   469 Aborted                 (core dumped) /usr/bin/mpv --playlist=playlist
```

### Log file

* mpv [log file](https://0x0.st/zxVy.log)
* `WAYLAND_DEBUG=1` [log file](https://0x0.st/zxVv.log)

### Sample files

none required.

unfortunately, after getting the PKGBUILD to work, I am getting an error during build.

the error doesn't appear when building from the repository though.

```../video/out/opengl/hwdec_vaegl.c: In function 'mapper_map':
../video/out/opengl/hwdec_vaegl.c:373:24: error: 'EGL_DMA_BUF_PLANE1_FD_EXT' undeclared (first use in this function); did you mean 'EGL_DMA_BUF_PLANE0_FD_EXT'?
             ADD_ATTRIB(EGL_DMA_BUF_PLANE ## plane ## _FD_EXT, \
                        ^~~~~~~~~~~~~~~~~
../video/out/opengl/hwdec_vaegl.c:334:31: note: in definition of macro 'ADD_ATTRIB'
     attribs[num_attribs++] = (name);                    \
                               ^~~~
../video/out/opengl/hwdec_vaegl.c:383:13: note: in expansion of macro 'ADD_PLANE_ATTRIBS'
             ADD_PLANE_ATTRIBS(1);
             ^~~~~~~~~~~~~~~~~
../video/out/opengl/hwdec_vaegl.c:373:24: note: each undeclared identifier is reported only once for each function it appears in
             ADD_ATTRIB(EGL_DMA_BUF_PLANE ## plane ## _FD_EXT, \
                        ^~~~~~~~~~~~~~~~~
../video/out/opengl/hwdec_vaegl.c:334:31: note: in definition of macro 'ADD_ATTRIB'
     attribs[num_attribs++] = (name);                    \
                               ^~~~
../video/out/opengl/hwdec_vaegl.c:383:13: note: in expansion of macro 'ADD_PLANE_ATTRIBS'
             ADD_PLANE_ATTRIBS(1);
             ^~~~~~~~~~~~~~~~~
../video/out/opengl/hwdec_vaegl.c:375:24: error: 'EGL_DMA_BUF_PLANE1_OFFSET_EXT' undeclared (first use in this function); did you mean 'EGL_DMA_BUF_PLANE0_OFFSET_EXT'?
             ADD_ATTRIB(EGL_DMA_BUF_PLANE ## plane ## _OFFSET_EXT, \
                        ^~~~~~~~~~~~~~~~~
../video/out/opengl/hwdec_vaegl.c:334:31: note: in definition of macro 'ADD_ATTRIB'
     attribs[num_attribs++] = (name);                    \
                               ^~~~
../video/out/opengl/hwdec_vaegl.c:383:13: note: in expansion of macro 'ADD_PLANE_ATTRIBS'
             ADD_PLANE_ATTRIBS(1);
             ^~~~~~~~~~~~~~~~~
../video/out/opengl/hwdec_vaegl.c:377:24: error: 'EGL_DMA_BUF_PLANE1_PITCH_EXT' undeclared (first use in this function); did you mean 'EGL_DMA_BUF_PLANE0_PITCH_EXT'?
             ADD_ATTRIB(EGL_DMA_BUF_PLANE ## plane ## _PITCH_EXT, \
                        ^~~~~~~~~~~~~~~~~
../video/out/opengl/hwdec_vaegl.c:334:31: note: in definition of macro 'ADD_ATTRIB'
     attribs[num_attribs++] = (name);                    \
                               ^~~~
../video/out/opengl/hwdec_vaegl.c:383:13: note: in expansion of macro 'ADD_PLANE_ATTRIBS'
             ADD_PLANE_ATTRIBS(1);
             ^~~~~~~~~~~~~~~~~
../video/out/opengl/hwdec_vaegl.c:373:24: error: 'EGL_DMA_BUF_PLANE2_FD_EXT' undeclared (first use in this function); did you mean 'EGL_DMA_BUF_PLANE0_FD_EXT'?
             ADD_ATTRIB(EGL_DMA_BUF_PLANE ## plane ## _FD_EXT, \
                        ^~~~~~~~~~~~~~~~~
../video/out/opengl/hwdec_vaegl.c:334:31: note: in definition of macro 'ADD_ATTRIB'
     attribs[num_attribs++] = (name);                    \
                               ^~~~
../video/out/opengl/hwdec_vaegl.c:385:13: note: in expansion of macro 'ADD_PLANE_ATTRIBS'
             ADD_PLANE_ATTRIBS(2);
             ^~~~~~~~~~~~~~~~~
../video/out/opengl/hwdec_vaegl.c:375:24: error: 'EGL_DMA_BUF_PLANE2_OFFSET_EXT' undeclared (first use in this function); did you mean 'EGL_DMA_BUF_PLANE0_OFFSET_EXT'?
             ADD_ATTRIB(EGL_DMA_BUF_PLANE ## plane ## _OFFSET_EXT, \
                        ^~~~~~~~~~~~~~~~~
../video/out/opengl/hwdec_vaegl.c:334:31: note: in definition of macro 'ADD_ATTRIB'
     attribs[num_attribs++] = (name);                    \
                               ^~~~
../video/out/opengl/hwdec_vaegl.c:385:13: note: in expansion of macro 'ADD_PLANE_ATTRIBS'
             ADD_PLANE_ATTRIBS(2);
             ^~~~~~~~~~~~~~~~~
../video/out/opengl/hwdec_vaegl.c:377:24: error: 'EGL_DMA_BUF_PLANE2_PITCH_EXT' undeclared (first use in this function); did you mean 'EGL_DMA_BUF_PLANE0_PITCH_EXT'?
             ADD_ATTRIB(EGL_DMA_BUF_PLANE ## plane ## _PITCH_EXT, \
                        ^~~~~~~~~~~~~~~~~
../video/out/opengl/hwdec_vaegl.c:334:31: note: in definition of macro 'ADD_ATTRIB'
     attribs[num_attribs++] = (name);                    \
                               ^~~~
../video/out/opengl/hwdec_vaegl.c:385:13: note: in expansion of macro 'ADD_PLANE_ATTRIBS'
             ADD_PLANE_ATTRIBS(2);
             ^~~~~~~~~~~~~~~~~
../video/out/opengl/hwdec_vaegl.c:373:24: error: 'EGL_DMA_BUF_PLANE3_FD_EXT' undeclared (first use in this function); did you mean 'EGL_DMA_BUF_PLANE0_FD_EXT'?
             ADD_ATTRIB(EGL_DMA_BUF_PLANE ## plane ## _FD_EXT, \
                        ^~~~~~~~~~~~~~~~~
../video/out/opengl/hwdec_vaegl.c:334:31: note: in definition of macro 'ADD_ATTRIB'
     attribs[num_attribs++] = (name);                    \
                               ^~~~
../video/out/opengl/hwdec_vaegl.c:387:13: note: in expansion of macro 'ADD_PLANE_ATTRIBS'
             ADD_PLANE_ATTRIBS(3);
             ^~~~~~~~~~~~~~~~~
../video/out/opengl/hwdec_vaegl.c:375:24: error: 'EGL_DMA_BUF_PLANE3_OFFSET_EXT' undeclared (first use in this function); did you mean 'EGL_DMA_BUF_PLANE0_OFFSET_EXT'?
             ADD_ATTRIB(EGL_DMA_BUF_PLANE ## plane ## _OFFSET_EXT, \
                        ^~~~~~~~~~~~~~~~~
../video/out/opengl/hwdec_vaegl.c:334:31: note: in definition of macro 'ADD_ATTRIB'
     attribs[num_attribs++] = (name);                    \
                               ^~~~
../video/out/opengl/hwdec_vaegl.c:387:13: note: in expansion of macro 'ADD_PLANE_ATTRIBS'
             ADD_PLANE_ATTRIBS(3);
             ^~~~~~~~~~~~~~~~~
../video/out/opengl/hwdec_vaegl.c:377:24: error: 'EGL_DMA_BUF_PLANE3_PITCH_EXT' undeclared (first use in this function); did you mean 'EGL_DMA_BUF_PLANE0_PITCH_EXT'?
             ADD_ATTRIB(EGL_DMA_BUF_PLANE ## plane ## _PITCH_EXT, \
                        ^~~~~~~~~~~~~~~~~
../video/out/opengl/hwdec_vaegl.c:334:31: note: in definition of macro 'ADD_ATTRIB'
     attribs[num_attribs++] = (name);                    \
                               ^~~~
../video/out/opengl/hwdec_vaegl.c:387:13: note: in expansion of macro 'ADD_PLANE_ATTRIBS'
             ADD_PLANE_ATTRIBS(3);
             ^~~~~~~~~~~~~~~~~

Waf: Leaving directory `/home/alarm/mpv-pkgbuild/src/mpv-0.29.1/build'
Build failed
 -> task in 'objects' failed with exit status 1 (run with -v to display more information)
==> ERROR: A failure occurred in build().
    Aborting...```

--- a/video/out/wayland_common.c
+++ b/video/out/wayland_common.c
@@ -1085,10 +1085,10 @@ int vo_wayland_init(struct vo *vo)
     if (create_xdg_surface(wl))
         return false;
 
-    if (wl->dnd_devman) {
+    if (wl->dnd_devman && wl->seat) {
         wl->dnd_ddev = wl_data_device_manager_get_data_device(wl->dnd_devman, wl->seat);
         wl_data_device_add_listener(wl->dnd_ddev, &data_device_listener, wl);
-    } else {
+    } else if (!wl->dnd_devman) {
         MP_VERBOSE(wl, "Compositor doesn't support the %s (ver. 3) protocol!\n",
                    wl_data_device_manager_interface.name);
     }

